float64 position_x
float64 position_y
float64 vel_x
float64 vel_y